src/components/BulkOperations/index.jsx

```diff
@@ -0,0 +1,208 @@
+import { useEffect, useState } from 'react';
+import {Link} from 'react-router-dom'
+import classNames from 'classnames';
+import Papa from 'papaparse';
+import {
+  SelectMenu, MenuItem, Dropzone, DataTable, Alert,
+} from '@openedx/paragon';
+
+import Collapsible from '../Collapsible';
+import LoadingSpinner from '../LoadingSpinner';
+import { formatDate } from '../../utils';
+import DiscoveryDataApiService from '../../data/services/DiscoveryDataApiService';
+import './BulkOperations.scss';
+
+const availableOperations = {
+  course_create: 'Bulk Create',
+  course_partial_update: 'Bulk Course Update',
+  course_run_partial_update: 'Bulk CourseRun Update',
+  course_rerun: 'Bulk Rerun',
+};
+
+const BulkOperations = () => {
+  const [bulkOperationId, setBulkOperationId] = useState(null);
+  const [fileContent, setFileContent] = useState(null);
+  const [file, setFile] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [submitSuccess, setSubmitSuccess] = useState(null);
+  const [historicalRecords, setHistoricalRecords] = useState(null);
+
+  function filteredHistoricalRecords() {
+    if (historicalRecords == null) {
+      return [];
+    }
+    return historicalRecords.filter(rec => !bulkOperationId || rec.task_type === bulkOperationId);
+  }
+
+  function handleUploadNew() {
+    setFileContent(null);
+    setFile(null);
+  }
+
+  function handleFileUpload({ fileData }) {
+    const file = fileData.get('file');
+    const reader = new FileReader();
+    reader.onload = (e) => {
+      const text = e.target.result;
+      setFile(file);
+      setFileContent(text);
+    };
+    reader.readAsText(file);
```

Comment:
why is this call needed?

Reply:
To read the file

Reply:
The onload is fired when the reader has read the file. To tell the reader to read the file, we call readAsText(fileToRead)

Reply:
The FileReader [API](https://developer.mozilla.org/en-US/docs/Web/API/FileReader)